lib/resources/datasets.js

```diff
@@ -20,21 +20,22 @@ module.exports = (service, endpoint) => {
       .then((project) => auth.canOrReject('dataset.list', project))
       .then(() => Datasets.getAllByProjectId(params.id, queryOptions))));
 
-  service.get('/projects/:projectId/datasets/:name', endpoint(({ Datasets, Projects }, { params, auth }) =>
-    Projects.getById(params.projectId)
+  service.get('/projects/:projectId/datasets/:name', endpoint(({ Datasets }, { params, auth }) =>
+    Datasets.getByProjectAndName(params.projectId, params.name)
       .then(getOrNotFound)
-      .then((project) => auth.canOrReject('dataset.read', project))
-      .then(() => Datasets.getDatasetMetadata(params.name, params.projectId)
-        .then(getOrNotFound))));
+      .then((dataset) => auth.canOrReject('dataset.read', dataset))
+      .then(() => Datasets.getDatasetMetadata(params.name, params.projectId))));
```

Comment:
`getDatasetMetadata` contains another call to `getByProjectAndName` that we can find a way to optimize out - can pass the dataset frame directly to this function. 

Another option:
Data file for dataset can be responsible for compositing things together: dataset, properties, linked forms. 
Query module can just be response for sql to fetch this info. 
Resource here can also stay simple.